apps/src/tests/TestPR3045.tsx

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import { NavigationContainer, ParamListBase } from '@react-navigation/native';
+import { NativeStackNavigationProp, createNativeStackNavigator } from '@react-navigation/native-stack';
+import { Button, ScrollView } from 'react-native';
+import { Text } from 'react-native-gesture-handler';
+
+type StackRouteParamList = {
+  First: undefined;
+  Second: undefined;
+  Third: undefined;
+};
+
+type NavigationProp<ParamList extends ParamListBase> = {
+  navigation: NativeStackNavigationProp<ParamList>;
+};
+
+type StackNavigationProp = NavigationProp<StackRouteParamList>;
+
+const First = ({navigation}: StackNavigationProp) => {
+  return <ScrollView contentInsetAdjustmentBehavior='automatic'>
+    <Button onPress={() => navigation.navigate("Second")} title='Second'/>
+    <Button onPress={() => navigation.preload("Second")} title='Preload Second'/>
+    <Button onPress={() => navigation.preload("Third")} title='Preload Third'/>
+  </ScrollView>
+}
```

Comment:
Please use a (regular) function here. This is my preference, but I try to enforce it across the repo, since regular functions take part in symbol name hoisting, while arrow functions do not.